### PR TITLE
APIS-6294 - Adding the connection to APM

### PR DESCRIPTION
**Pull request by Tobias1087**

### CODE REVIEW

File: app/uk/gov/hmrc/apigatekeeperapisfrontend/views/ApiListPage.scala.html [hunk 1]
```diff
@@ -15,13 +15,30 @@
...
 
-@()(implicit request: LoggedInRequest[_])
+@(apis: List[ApiData])(implicit request: LoggedInRequest[_])
 
     @layout(pageTitle = Some("APIs")){
       <h1 class="govuk-heading-l">
           APIs
       </h1>
-    }
+    <table class="govuk-table responsive-table" data-module="moj-sortable-table">
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapisfrontend/views/ApiListPage.scala.html]: Did you consider using the Twirl component for the table? See [example](https://github.com/hmrc/play-frontend-govuk-examples/blob/main/src/test/play-2/twirl/uk/gov/hmrc/govukfrontend/views/examples/table/columnWidths.scala.html).

File: test-common/uk/gov/hmrc/apigatekeeperapisfrontend/utils/ApiDataTestData.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  def anApiDataMap(): ApiData.ApiDefinitionMap = {
+
+    Map(defaultContext -> ApiData(
+      ServiceName("helloworld"),
+      "a.com/",
+      "Hello World",
+      "This is a test api",
+      defaultContext,
+      Map(defaultVersion -> ApiVersion(defaultVersion, ApiStatus.STABLE, ApiAccess.PUBLIC, List(), true, None, ApiVersionSource.OAS)),
+      false,
```
Comment (johnsgp) [inline test-common/uk/gov/hmrc/apigatekeeperapisfrontend/utils/ApiDataTestData.scala]: Having some argument names here would be helpful for readability.

File: test/uk/gov/hmrc/apigatekeeperapisfrontend/services/ApmServiceTest.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "use production if only production returned" in new Setup {
+      ApmConnectorMock.returnsNoData(Environment.SANDBOX)
+      ApmConnectorMock.returnsData(Environment.PRODUCTION)
+      val result = await(service.fetchAllApis())
+      result.size shouldBe 1
+      result.get(defaultContext) shouldBe anApiDataMap().get(defaultContext)
+    }
+
+    "merge them when both are returned" in new Setup {
+      ApmConnectorMock.returnsData(Environment.SANDBOX)
```
Comment (johnsgp) [inline test/uk/gov/hmrc/apigatekeeperapisfrontend/services/ApmServiceTest.scala]: Feels like this test should include at least 2 APIs in the result.

File: test/uk/gov/hmrc/apigatekeeperapisfrontend/services/ApmServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "use production if only production returned" in new Setup {
+      ApmConnectorMock.returnsNoData(Environment.SANDBOX)
+      ApmConnectorMock.returnsData(Environment.PRODUCTION)
+      val result = await(service.fetchAllApis())
+      result.size shouldBe 1
+      result.get(defaultContext) shouldBe anApiDataMap().get(defaultContext)
+    }
+
+    "merge them when both are returned" in new Setup {
+      ApmConnectorMock.returnsData(Environment.SANDBOX, anApiDataMap().concat(Map(ApiContext("sandbox") -> defaultData.copy(context = ApiContext("sandbox")))))
```
Comment (johnsgp) [inline test/uk/gov/hmrc/apigatekeeperapisfrontend/services/ApmServiceSpec.scala]: Using `sandbox` and `production` as contexts is confusing. Could these be more context-like?

**Review by johnsgp — APPROVED** [review]

**Review by GurpreetBhamra — APPROVED** [review]

**Pull request merged**